File: backend/src/Microservices/PackageDelivery/PackageDelivery.DAL/Entities/VehicleUsage.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TypeGen.Core.TypeAnnotations;

namespace PackageDelivery.DAL.Entities
{
    [ExportTsInterface(OutputDir = "../../../../../frontend/web/parceldelivery-admin-app/src/app/_models")]
    public class VehicleUsage
    {
        [BsonId]
        [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
        public string Id { get; set; }
        public string EmployeeId { get; set; }
        public string EmployeeName { get; set; }
        public string EmployeeEmail { get; set; }
        public Vehicle Vehicle { get; set; }
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
        public string Note { get; set; }
    }
}

File: frontend/mobil/parceldelivery_mobile/lib/models/vehicle_usage.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:parceldelivery_mobile/models/vehicle.dart';

part 'vehicle_usage.freezed.dart';
part 'vehicle_usage.g.dart';

@freezed
class VehicleUsage with _$VehicleUsage {
  const factory VehicleUsage({
    required String id,
    required String employeeId,
    required String employeeName,
    required String employeeEmail,
    required Vehicle vehicle,
    required DateTime dateFrom,
    required DateTime dateTo,
    required String note,
  }) = _VehicleUsage;

  factory VehicleUsage.fromJson(Map<String, Object?> json) =>
      _$VehicleUsageFromJson(json);
}
